domain/consensus/utils/serialization/common.go

```diff
@@ -19,7 +19,11 @@ func WriteElement(w io.Writer, element interface{}) error {
 	// type assertions first.
 	switch e := element.(type) {
 	case []byte:
-		_, err := w.Write(e)
+		err := WriteElement(w, len(e))
```

Comment:
We don't have support for `int`, so this won't work, can you cast to `uint64`? (or just write it as uint64 here)